GUIs/ScriptRunnerPyQt5_GUI/scripFunc/scripUNIMULTI.py

```diff
@@ -740,28 +748,47 @@ def OUTPUTBOX(text):
                     OUTPUT.exec_()
 
                 def hashchk(hashnumber):
-                    linhashtypes = ['MD5', 'sha1sum', 'sha256sum', 'sha384sum', 'sha512sum']
-                    winhashtypes = ['MD5', 'SHA1', 'SHA256', 'SHA384', 'SHA512']
-
-                    if platform == 'linux' or platform == 'darwin':
-                        filepath = self.fpath.text()
-                        command = r'sudo ' + linhashtypes[hashnumber] + ' ' + filepath
-                        EXEC = sub.Popen(command.split(), stdout=sub.PIPE)
-                        stdout, _ = EXEC.communicate()
-                        output = stdout.decode("utf-8")
-                        OUTPUTBOX(output)
-                    elif platform == 'win32':
-                        print(winhashtypes[hashnumber])
-                        filepath = self.fpath.text()
-                        command = r'Get-Filehash ' + filepath + ' -Algorithm ' + winhashtypes[
-                            hashnumber] + ' | Format-List'
-                        print(command)
-                        EXEC = sub.Popen(["powershell", "& {" + command + "}"], stdout=sub.PIPE)
-                        stdout, _ = EXEC.communicate()
-                        output = stdout.decode("utf-8")
-                        OUTPUTBOX(output)
-
-                self.genhash.clicked.connect(lambda: hashchk(self.hashnumber))
+                    if len(self.fpath.text()) != 0 and Path(self.fpath.text()).is_file():
+                        linhashtypes = ['MD5', 'sha1sum', 'sha256sum', 'sha384sum', 'sha512sum']
+                        winhashtypes = ['MD5', 'SHA1', 'SHA256', 'SHA384', 'SHA512']
+
+                        if platform == 'linux' or platform == 'darwin':
+                            filepath = self.fpath.text()
+                            command = r'sudo ' + linhashtypes[hashnumber] + ' ' + filepath
+                            EXEC = sub.Popen(command.split(), stdout=sub.PIPE)
+                            stdout, _ = EXEC.communicate()
+                            output = stdout.decode("utf-8")
+                            OUTPUTBOX(output)
+                        elif platform == 'win32':
+                            print(winhashtypes[hashnumber])
+                            filepath = self.fpath.text()
+                            command = r"Get-Filehash '" + filepath + "' -Algorithm " + winhashtypes[
+                                hashnumber] + " | Format-List"
+                            print(command)
+                            EXEC = sub.Popen(["powershell", "& {" + command + "}"], stdout=sub.PIPE)
+                            stdout, _ = EXEC.communicate()
+                            output = stdout.decode("utf-8")
+                            OUTPUTBOX(output)
+                    elif len(self.fpath.text()) == 0:
+                        print('No file path entered')
+                        ERROR_NO_FILEPATH = QMessageBox()
```

Comment:
Codacy found an issue: ['QMessageBox' may be undefined, or defined from star imports: PyQt5.QtWidgets (F405)](https://app.codacy.com/manual/vipersniper0501/CP_Scripts2/pullRequest?prid=5927260)